### PR TITLE
Support for Full app conversion attribution and deep linking

### PR DESCRIPTION
**Pull request by sojanpr**
Support for Full app conversion attribution and deep linking
@aaustin @EvangelosG @Sarkar @dmitrig01 @derrickstaten

### CODE REVIEW

File: Branch-SDK/src/io/branch/referral/Branch.java [hunk 1]
```diff
@@ -291,8 +291,11 @@
 
     private static boolean isLogging_ = false;
 
-    private static boolean checkInstallReferrer_ = false;
-    private static long PLAYSTORE_REFERRAL_FETCH_WAIT_FOR = 5000;
+    // Branch will wait for 1.5 sec to capture the install referrer inorder to provide
+    // accurate attribution . This will be only on the first time user open the app.
+    // @see to skip this please call {@link #disablePlayStoreReferrer() }
+    private static boolean checkInstallReferrer_ = true;
```
Comment (Sarkar) [inline Branch-SDK/src/io/branch/referral/Branch.java]: my only concern with this PR is setting this default to true, what are your thoughts @sojanpr

Comment (EvangelosG) [inline Branch-SDK/src/io/branch/referral/Branch.java]: I'm conflicted as well @Sarkar -- it's probably one of those things devs/users are never going to notice, but as a performance queen I would not want *any* SDK by default taking 1.5 seconds extra to gather something that 1) might not show up in time (1.5 seconds seems like it won't?) 2) I might not care about.

Comment (sojanpr) [inline Branch-SDK/src/io/branch/referral/Branch.java]: This is only for Install  and that happens only one time. Subsequent open will not have  this delay. We have option to reset it for any one who is worried  about this deal over accurate tracking.

Comment (EvangelosG) [inline Branch-SDK/src/io/branch/referral/Branch.java]: Yeah I saw it's only for installs, that's when you want the user to have the least delayed experience possible IMO. In my metric lacking opinion, playstore referrer information is a less utilized/more advanced feature(?). If you think **every** dev can benefit from this feature then it should be on, but our on-boarding/integration docs will have to cover it, I can already see the tickets in support "why does my app take longer to boot up on install???? I only want deep linking, why does it have to wait for something I'M NOT USING????!?!"

**Comment by derrickstaten** [comment]
Per my comment in slack, it'd be great to have the Android and iOS functionality match up. On iOS, partners can specify the delay themselves (it's not fixed to 1.5 seconds). See https://github.com/BranchMetrics/ios-branch-deep-linking/blob/master/Branch-SDK/Branch-SDK/Branch.h#L519

File: Branch-SDK/src/io/branch/referral/Branch.java [hunk 2]
```diff
@@ -459,6 +459,9 @@ public void setDebug() {
         enableTestMode();
     }
 
+    // Since play store referrer broadcast from google is few millisecond delayed, delay Branch init by calling this method for more accurate
+    // tracking and attribution. This will be only on the first time user open the app.
+    // Note : This method should be called in order to handle IA - Full app conversion deeplinking and attribution
```
Comment (aaustin) [inline Branch-SDK/src/io/branch/referral/Branch.java]: Let's add another note and please update with correctness:
`// Note: Recommend 1500 to capture more than 90% of the install referrer cases per our testing as of 4/2017`

**Comment by aaustin** [comment]
Almost there. Let's just add the guidance on number to wait for

**Comment by aaustin** [comment]
Looks good 👍

**Pull request merged**